reflex/ivars/number.py

```diff
@@ -20,13 +21,33 @@
     var_operation_return,
 )
 
-NUMBER_T = TypeVar("NUMBER_T", int, float, Union[int, float])
+NUMBER_T = TypeVar("NUMBER_T", int, float, Union[int, float], bool)
+
+if TYPE_CHECKING:
+    from .sequence import ArrayVar
+
+
+def raise_unsupported_operand_types(
+    operator: str, operands_types: tuple[type, ...]
+) -> NoReturn:
+    """Raise an unsupported operand types error.
+
+    Args:
+        operator: The operator.
+        operands_types: The types of the operands.
+
+    Raises:
+        VarTypeError: The operand types are unsupported.
+    """
+    raise VarTypeError(
```

Comment:
i think the proper thing to do in this scenario is to `return NotImplemented` [[1]](https://docs.python.org/3.6/library/constants.html#NotImplemented)

Not sure if that totally makes sense for us though, because ostensibly we're not expecting an incompatible type to have a better comparison checker than the Var itself, and raising the exception means we fail loud and fast (and record the exception type for telemetry).

Not suggesting a change here, just posing a point for discussion.

Reply:
that seems valid, we would need to also remove `.operation` from `Var` for both to work. Then Python should do the heavy lifting for us.